### ISSUE TITLE
Debug angular+typescript either using browser or VScode with jest

### ISSUE THREAD

**Issue by asadsahi** [thymikee/jest-preset-angular#132]
Can't find many resources online which show how to debug angular tests with jest. I am wondering if there is a way to test angula+typescript based tests (which in most case will be angular-cli based project).

**Comment by thymikee** [comment]
Take a look here for vscode setup: https://github.com/facebook/jest/tree/master/.vscode

**Comment by asadsahi** [comment]
@thymikee that isn't a typescript setup. I have updated title to clarify as well.

**Comment by thymikee** [comment]
Maybe this will help https://github.com/artsy/reaction/blob/master/.vscode/launch.json

**Comment by asadsahi** [comment]
@thymikee again that isn't an angular setup. I have tried using their launch.json config for debugging test but it isn't connecting. some of the settings in launch.json they are using is outdated tool. 

I have [this ](github.com/asadsahi/ngnode-fullstack) project where you can run test with `npm test` from command line. Would love to somehow debug these tests. would you mind having a look?

**Comment by thymikee** [comment]
Angular is just a TS project, so I don't see why this wouldn't work. Also, I don't usually need to do it, so I'm not a best fit to check it. But I'll leave the issue open.
cc @SimenB @orta as they have more hands-on experience on that

**Comment by orta** [comment]
Unless you're trying to get all the source mapping working, the reaction launch example should be enough to get you going. This is a general vscode + jest debugging issue, rather than anything specific to this preset.

**Comment by asadsahi** [comment]
@orta thanks for your response. reactions debug launch keeps on giving this error:
![image](https://user-images.githubusercontent.com/3470892/37101969-aa53b2d4-221e-11e8-81b3-669c296120be.png)

Note: I have to change port and debug options in runtimeArgs to satisfy current node debugging protocol:

![image](https://user-images.githubusercontent.com/3470892/37102063-df7fce2a-221e-11e8-8a3d-2a3f6bd299da.png)

**Comment by orta** [comment]
I'm afraid I don't have the time to help you figure it out, I'd recommend looking at the vscode documentation - https://code.visualstudio.com/Docs/editor/debugging and here's one I did last week in the jest codebase - https://github.com/facebook/jest/blob/master/.vscode/launch.json

**Comment by asadsahi** [comment]
@orta the question asked in this issue was Angular + Typescript + Jest debugging. Links you are sharing are for javascript + jest debugging in vs code. Unfortunately I couldn't find any clear instructions so far of configuring/debugging angular + typescript code with jest. 

Thanks for your help anyway.

**Comment by jfcere** [comment]
#### I thought this would worth sharing...

I've struggled a lot trying to make VsCode debugger to work with custom launch tasks to no avail ... wanted me to throw away Jest because it was a pain to debug! After searching for configurations on the web I fell on a similar issue on [Nrwl/nx](https://github.com/nrwl/nx) repository. A guy was talking about _vs-code jest_ extension that made his setup worked! I was skeptical but what did I have to lose?

#### TLDR: (you talk too much...)

I ended up ditching the `launch.json` configuration and installed [vscode-jest](https://marketplace.visualstudio.com/items?itemName=Orta.vscode-jest) extension.
Yeah, that's it that's all ... bingo banjo! 😄 

Tweeking a little bit the extension configuration in `.vscode/settings.json` I've found myself to be on cloud 9 and reconciled with Jest!

```json
// Enable debug mode to diagnose plugin issues. (see developer console)
"jest.debugMode": true,
// Whether codelens for debugging should show
"jest.enableCodeLens": true,
// Whether codelens for debugging should show
"jest.debugCodeLens.showWhenTestStateIn": [
  "fail",
  "unknown",
  "pass"
]
```
